ui/components/multichain/global-menu/global-menu.tsx

```diff
@@ -191,44 +195,70 @@ export const GlobalMenu = ({
       return;
     }
 
+    const currentEnvironment = getEnvironmentType();
+    const newValue = !isSidePanelDefault;
+
     try {
-      const newValue = !isSidePanelDefault;
-      await dispatch(setUseSidePanelAsDefault(newValue));
-
-      // If switching from sidepanel to popup view, close the current sidepanel
-      if (
-        isSidePanelDefault &&
-        getEnvironmentType() === ENVIRONMENT_TYPE_SIDEPANEL
-      ) {
-        // Close only the sidepanel, not the entire browser window
+      // Case 1: Currently in sidepanel, switching to popup
+      if (currentEnvironment === ENVIRONMENT_TYPE_SIDEPANEL) {
+        await dispatch(setUseSidePanelAsDefault(false));
+        // Close the sidepanel
         window.close();
+        return;
       }
-      // If switching from popup to sidepanel view, open the sidepanel
-      else if (
-        !isSidePanelDefault &&
-        getEnvironmentType() === ENVIRONMENT_TYPE_POPUP
-      ) {
-        try {
+
+      // Case 2: Currently in popup
+      if (currentEnvironment === ENVIRONMENT_TYPE_POPUP) {
+        // If switching TO sidepanel (newValue is true)
+        if (newValue) {
           const browserWithSidePanel = browser as typeof browser & {
             sidePanel?: {
               open: (options: { windowId: number }) => Promise<void>;
             };
           };
-          if (browserWithSidePanel?.sidePanel?.open) {
-            const tabs = await browser.tabs.query({
-              active: true,
-              currentWindow: true,
+
+          // If sidePanel.open is undefined (e.g., Arc browser), don't do anything
+          if (!browserWithSidePanel?.sidePanel?.open) {
+            console.log(
+              'sidePanel.open is undefined - browser does not support sidepanel',
+            );
+            return;
+          }
+
+          const tabs = await browser.tabs.query({
+            active: true,
+            currentWindow: true,
+          });
+
+          if (tabs && tabs.length > 0 && tabs[0].windowId) {
+            // Try to open sidepanel FIRST, before setting preference
+            await browserWithSidePanel.sidePanel.open({
+              windowId: tabs[0].windowId,
             });
-            if (tabs && tabs.length > 0 && tabs[0].windowId) {
-              await browserWithSidePanel.sidePanel.open({
-                windowId: tabs[0].windowId,
-              });
-              // Close the popup after successfully opening the sidepanel
-              window.close();
+
+            // Wait a moment and verify sidepanel actually opened
+            await new Promise((resolve) => setTimeout(resolve, 500));
+
+            // Check if sidepanel context exists - if not, the open didn't work (Arc browser)
+            const contexts = await chrome.runtime.getContexts({
+              contextTypes: ['SIDE_PANEL' as chrome.runtime.ContextType],
+            });
```

Comment:
[Nit] I get the need for this though wonder if we can polish it slightly. With this 500ms wait it means all users clicking Sidepanel from global menu see the menu linger for 500ms. Instead we could have a fn like (pseudocode):

```
const checkSideBarIsOpen = async () => {
  let count = 0

  while (count < 10) {
    if (isSideBarOpen()) {
      return true
    } else {
      await waitFor(100ms)
     count++
    }
  }

  return false
}
```

So that if the sidebar is open it finishes much sooner and could also wait for longer if it is not yet.